webui/src/components/RepositoryListPage.js

```diff
@@ -29,7 +30,28 @@ const CreateRepositoryModal = ({show, error, onSubmit, onCancel}) => {
     );
 };
 
-const RepositoryList = ({ list, paginate }) => {
+const RepositoryList = connect(
+    ({ repositories }) => ({ deleteStatus: repositories.delete }),
+    ({ deleteRepository, listRepositories })
+)(({ list, paginate, deleteStatus, deleteRepository, listRepositories}) => {
+
+    const [showingDeleteModal, setShowDeleteModal] = useState(false);
+    const closeDeleteModal = () => setShowDeleteModal(false);
+    const showDeleteModal = () => setShowDeleteModal(true);
```

Comment:
can you call `setShowDeleteModal` with true/false when needed, I think it will be clear enough and don't want to get the function population going up because of defaults.